### ISSUE TITLE
How to properly document a file response

### ISSUE THREAD

**Issue by adin234** [marshmallow-code/flask-smorest#142]
This doesn't look right to me

```python
@blp.route('/some-file-returning-endpoint')
class SomeResource(MethodView):
  @blp.doc(operationId='getFile')
  @blp.doc(**{ 'responses': { '200': { 'description': 'Zip file containing the model/s', 'content': {'application/zip' : { 'schema': { 'format': 'binary', 'type': 'string' }}}}}})
  def post(self):
    """Get file

    Get the file
    """
    # some logic here to get the file
    content = some_file
    return Response(content,
                    mimetype='application/zip',
                    headers={'Content-Disposition': 'attachment;filename=models.zip'})
```

**Comment by lafrech** [comment]
That's a good question. I don't have an answer out of my head. We might have to implement something.

**Comment by lafrech** [comment]
We could add a `response_file` or `file_response` decorator but we might be able to just add a `is_file` argument to `response` (yes, that's a bit ugly,...) and get away with it.

We'd also need to add an argument to specify the content type, overriding the default one. Consider making it plural for #146.

**Comment by lafrech** [comment]
Or maybe we allow a Field to be passed instead of a Schema. This would be even more generic.

I'd need to check the spec to see if that makes sense and we could include this in apispec if it does.

Then we can define a FileField that will generate the correct doc.

**Comment by lafrech** [comment]
This is solved by #330.

https://flask-smorest.readthedocs.io/en/latest/response.html#file-response

```py
@blp.route("/")
@blp.response(
    200, {"format": "binary", "type": "string"}, content_type="application/csv"
)
def func():
    csv_str = ...
    response = Response(csv_str, mimetype="text/csv")
    response.headers.set("Content-Disposition", "attachment", filename="file.csv")
    return response
```

**Comment by yarkinslav** [comment]
> This is solved by #330.
> 
> https://flask-smorest.readthedocs.io/en/latest/response.html#file-response
> 
> ```python
> @blp.route("/")
> @blp.response(
>     200, {"format": "binary", "type": "string"}, content_type="application/csv"
> )
> def func():
>     csv_str = ...
>     response = Response(csv_str, mimetype="text/csv")
>     response.headers.set("Content-Disposition", "attachment", filename="file.csv")
>     return response
> ```

@lafrech 
what is csv_str value here?

**Comment by lafrech** [comment]
`csv_str` is the CSV file as string. You might be allowed to pass a `StringIO`, I don't remember. See Flask docs.
